tests/models/test_core_model.py

```diff
@@ -64,4 +64,11 @@ def test_core_model(data, param_set, curve_fun, loss_fun):
     assert model.bounds.shape == (4 * (3 + 1), 2)
     ub = [b[1] for b in model.bounds]
     assert ub[:4] == [np.inf] * 4
-    assert ub[4:] == [1.0, 2.0, 3.0, 3.0] * 3
\ No newline at end of file
+    assert np.linalg.norm(np.array(ub[4:]) - np.array([1.0, 2.0, 3.0, 3.0] * 3)) < 1e-10
+
```

Comment:
@zhengp0 again please do check the `asserts` on bounds and x_init. thanks!

Reply:
Looks correct! Minor suggestion, if we could remove the number by the actual variable that will be great. For example
```
num_groups = len(data.groups)
num_fe = param_set.num_fe
x0 = np.array([0.0] * num_fe * (1 + num_groups))
model.objective(x0, data)

assert model.bounds.shape == (num_fe * (num_groups + 1), 2)
ub = [b[1] for b in model.bounds]
assert ub[:num_fe] == [np.inf] * num_fe
assert np.allclose(np.array(ub[num_fe:]), np.array([1.0, 2.0, 3.0, 3.0] * num_groups))
```

Reply:
yes, I changed that!

Reply:
great! Thanks!